### PR TITLE
Schema: Add User structured core kind

### PR DESCRIPTION
**Pull request by spinillos**
It adds schema for this DTO:
* It's the same model but its partially refactored into its own service. This is the DTO used in the response.
  * [services/user/model.go](https://github.com/grafana/grafana/blob/main/pkg/services/user/model.go#L142)
  * [models/user.go](https://github.com/grafana/grafana/blob/main/pkg/models/user.go#L111)

There are another ones into "user" like:
* [UserSearchHitDTO](https://github.com/grafana/grafana/blob/main/pkg/models/user.go#L128): That its using for the search and looks like another kind of user.

But we also have another services like sign up and login that they are using another information to create/login a user:
[api/dtos/user.go](https://github.com/grafana/grafana/blob/main/pkg/api/dtos/user.go), in that case, should we add `Password`, `Code`, etc fields here?

### CODE REVIEW

File: kinds/structured/user/user_kind.cue [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+				// AuthLabels is a list of authentication providers used (OAuth, SAML, LDAP...)
+				authLabels: [...string]
+				// CreatedAt indicates when the user was created.
+				createdAt: int64 @grafanamaturity(ToMetadata="sys")
+				// UpdatedAt indicates when the user was updated.
+				updatedAt: int64 @grafanamaturity(ToMetadata="sys")
+				// AvatarUrl is the user's avatar URL.
+				avatarUrl: string
+				// AccessControl metadata associated with a given resource.
+				accessControl: [string]: bool @grafanamaturity(ToMetadata="sys")
```
Comment (spinillos) [inline kinds/structured/user/user_kind.cue:36]: For any reason, I cannot set this value as optional. The DTO [here](https://github.com/grafana/grafana/blob/main/pkg/services/user/model.go#L156) has the `omitempty` tag and adding the `?` should work. 

Is it a bug or is it intentional? @sdboyer

Comment (sdboyer) [reply kinds/structured/user/user_kind.cue:36]: `omitempty` has some weirdness to it - it doesn't actually correspond to optionality. The only way Go can really express field optionality is through a (nil) pointer. Some more discussion about that [here](https://github.com/grafana/grafana/pull/59621#discussion_r1037424816), and grafana/grok#1.

But when you say "adding the `?` should work, do you mean like this?

```cue
accessControl?: [string]: bool
```

Comment (spinillos) [reply kinds/structured/user/user_kind.cue:36]: Yes, I mean that.

Comment (sdboyer) [reply kinds/structured/user/user_kind.cue:36]: OK, _that_ may be hitting yet a different known-but-not-yet-clearly-documented issue - i believe that we've encountered an issue where combining an optional field with a map has unexpected results.

**Comment by joanlopez** [comment]
Hi @spinillos, @sdboyer, 

Is there any specific issue (perhaps https://github.com/grafana/thema/issues/100 or https://github.com/grafana/schematization-and-as-code-project/issues/32) preventing us from getting this merged?

Just trying to understand if I can help with trying to get this unstuck.

Thanks!

**Comment by spinillos** [comment]
I had less conflicts doing it again here: https://github.com/grafana/grafana/pull/63493

**Pull request closed without merging**